packages/astro/src/container/index.ts

```diff
@@ -0,0 +1,443 @@
+import type {
+	AstroConfig,
+	ComponentInstance,
+	MiddlewareHandler,
+	RouteData,
+	RouteType,
+	SSRLoadedRenderer,
+	SSRManifest,
+	SSRResult,
+	AstroUserConfig, 
+	AstroRenderer,
+} from '../@types/astro.js';
+import { TestPipeline } from './pipeline.js';
+import { Logger } from '../core/logger/core.js';
+import { nodeLogDestination } from '../core/logger/node.js';
+import type { SSRManifestI18n } from '../core/app/types.js';
+import { toRoutingStrategy } from '../i18n/utils.js';
+import { validateConfig } from '../core/config/config.js';
+import { ASTRO_CONFIG_DEFAULTS } from '../core/config/schema.js';
+import { RenderContext } from '../core/render-context.js';
+import { posix } from 'node:path';
+import { getParts, getPattern, validateSegment } from '../core/routing/manifest/create.js';
+import { removeLeadingForwardSlash } from '../core/path.js';
+import type {AstroComponentFactory} from "../runtime/server/index.js";
+
+/**
+ * Options to be passed when rendering a route
+ */
+export type ContainerRenderOptions = {
+	/**
+	 * If your component renders slots, that's where you want to fill the slots.
+	 * A single slot should have the `default` field:
+	 * 
+	 * ## Examples
+	 * 
+	 * **Default slot**
+	 * 
+	 * ```js
+	 * container.renderToString(Component, { slots: { default: "Some value"}});
+	 * ```
+	 * 
+	 * **Named slots**
+	 *
+	 * ```js
+	 * container.renderToString(Component, { slots: { "foo": "Some value", "bar": "Lorem Ipsum" }});
+	 * ```
+	 */
+	slots?: Record<string, any>;
+	/**
+	 * The request is used to understand which path/URL the component is about to render.
+	 *
+	 * Use this option in case your component or middleware needs to read information like `Astro.url` or `Astro.request`.
+	 */
+	request?: Request;
+	/**
+	 * Useful for dynamic routes. If your component is something like `src/pages/blog/[id]/[...slug]`, you'll want to provide:
+	 * ```js
+	 * container.renderToString(Component, { params: ["id", "...slug"] });
+	 * ```
+	 */
+	params?: Record<string, string | undefined>;
+	/**
+	 * Useful if your component needs to access some locals without the use a middleware.
+	 * ```js
+	 * container.renderToString(Component, { locals: { getSomeValue() {} } });
+	 * ```
+	 */
+	locals?: App.Locals;
+	/**
+	 * Useful in case you're attempting to render an endpoint:
+	 * ```js
+	 * container.renderToString(Endpoint, { routeType: "endpoint" });
+	 * ```
+	 */
+	routeType?: RouteType;
+};
+
+function createContainerManifest(
```

Comment:
is this naming right? it looks like it creates an SSR Manifest *from* a container manifest (among other things).

Reply:
Would `createManifest` be a better name?